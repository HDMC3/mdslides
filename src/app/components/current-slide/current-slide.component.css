:host {
    --zoom: 0.3;
    width: 100%;
    height: 100%;
    border: none;
    position: relative;
    overflow: hidden;
    animation: switchPanel 0.15s ease-in-out;
}

.current-slide-container {
    width: 100%;
    height: 100%;
    position: relative;
    overflow: scroll auto;
    background-color: var(--background-basic-color-1);
}

.zoom-buttons {
    position: absolute;
    bottom: 1.5rem;
    left: 1.5rem;
    display: flex;
    gap: 0.5rem
}

.top-options-container {
    position: absolute;
    top: 0;
    left: 0;
    padding: 0.5rem 0.7rem;
    display: flex;
    flex-wrap: wrap;
    gap: 0.2rem;
    justify-content: space-between;
    width: 100%;
    background-color: var(--background-basic-color-1);
}

.slide-name-container {
    display: flex;
    align-items: center;
    gap: 0.2rem;
}

.slide-name-control {
    display: flex;
    flex-direction: column;
}


@media screen and (min-width: 500px) {
    :host {
        --zoom: 0.4;
        height: 100%;
        width: 50%;
        border: none;
        display: initial;
    }

    .zoom-buttons {
        left: unset;
        right: 1.5rem;
    }
}