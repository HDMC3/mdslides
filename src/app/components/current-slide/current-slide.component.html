<div class="current-slide-container"></div>

<div class="top-options-container">
    <div *ngIf="!editingSlideName" class="slide-name-container">
        <button nbButton ghost shape="round" (click)="editSlideName()">
            <nb-icon icon="edit-outline"></nb-icon>
        </button>
        <h6 style="margin: 0;">{{ currentSlide?.name | maxLength: 30 }}</h6>
    </div>
    <div *ngIf="editingSlideName" class="slide-name-container">
        <button nbButton ghost shape="round" (click)="saveSlideName()">
            <nb-icon icon="save-outline"></nb-icon>
        </button>
        <div class="slide-name-control">
            <input nbInput type="text" fieldSize="small" autocomplete="off"
                [status]="slideNameFormControl.invalid ? 'danger' : 'basic'" 
                [formControl]="slideNameFormControl"
                (keyup.enter)="saveSlideName()">
            <small *ngIf="slideNameFormControl.invalid" style="color: var(--color-danger-500)">No pueder se vacio</small>
        </div>
    </div>
</div>


<div class="zoom-buttons">
    <button nbButton size="medium" outline (click)="zoomSlide('min')">
        <nb-icon icon="minimize-outline"></nb-icon>
    </button>
    <button nbButton size="medium" outline (click)="zoomSlide('max')">
        <nb-icon icon="maximize-outline"></nb-icon>
    </button>
</div>