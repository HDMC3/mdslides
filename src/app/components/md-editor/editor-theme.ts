import { EditorView } from '@codemirror/view';

const tooltipBackground = '#353a42';

const textBasicColor = 'var(--text-basic-color)';
const textHintColor = 'var(--text-hint-color)';
const backgroundBasicColor1 = 'var(--background-basic-color-1)';
const backgroundBasicColor2 = 'var(--background-basic-color-2)';
const backgroundBasicColor4 = 'var(--background-basic-color-4)';
const colorBasic800 = 'var(--color-basic-800)';
const colorBasicTransparent600 = 'var(--color-basic-transparent-600)';
const colorControlDefault = 'var(--color-control-default)';
const colorControlActive = 'var(--color-control-active)';
const colorControlHover = 'var(--color-control-hover)';

export const oneDarkTheme = EditorView.theme({
    '&': {
        color: textBasicColor,
        backgroundColor: backgroundBasicColor1
    },

    '.cm-content': {
        caretColor: textHintColor
    },

    '.cm-cursor, .cm-dropCursor': { borderLeftColor: textHintColor },
    '&.cm-focused .cm-selectionBackground, .cm-selectionBackground, .cm-content ::selection': { backgroundColor: colorBasicTransparent600 },

    '.cm-panels': { backgroundColor: backgroundBasicColor4, color: textBasicColor },
    '.cm-panels.cm-panels-top': { borderBottom: '2px solid black' },
    '.cm-panels.cm-panels-bottom': { borderTop: '2px solid black' },

    '.cm-button': { border: 'none', backgroundImage: 'none', background: colorControlDefault, color: colorBasic800 },
    '.cm-button:active': { border: 'none', backgroundImage: 'none', background: colorControlActive },
    '.cm-button:hover': { border: 'none', backgroundImage: 'none', background: colorControlHover },
    '.cm-textfield': { color: colorBasic800 },

    '.cm-panel button[name=close]': { color: textBasicColor },

    '.cm-searchMatch': {
        backgroundColor: '#72a1ff59',
        outline: '1px solid #457dff'
    },
    '.cm-searchMatch.cm-searchMatch-selected': {
        backgroundColor: '#6199ff2f'
    },

    '.cm-activeLine': { backgroundColor: '#6699ff0b' },
    '.cm-selectionMatch': { backgroundColor: '#aafe661a' },

    '&.cm-focused .cm-matchingBracket, &.cm-focused .cm-nonmatchingBracket': {
        backgroundColor: '#bad0f847',
        outline: '1px solid #515a6b'
    },

    '.cm-gutters': {
        backgroundColor: backgroundBasicColor2,
        color: textHintColor,
        border: 'none'
    },

    '.cm-activeLineGutter': {
        backgroundColor: backgroundBasicColor4
    },

    '.cm-foldPlaceholder': {
        backgroundColor: 'transparent',
        border: 'none',
        color: '#ddd'
    },

    '.cm-tooltip': {
        border: 'none',
        backgroundColor: backgroundBasicColor4
    },
    '.cm-tooltip .cm-tooltip-arrow:before': {
        borderTopColor: 'transparent',
        borderBottomColor: 'transparent'
    },
    '.cm-tooltip .cm-tooltip-arrow:after': {
        borderTopColor: tooltipBackground,
        borderBottomColor: tooltipBackground
    },
    '.cm-tooltip-autocomplete': {
        '& > ul > li[aria-selected]': {
            backgroundColor: colorBasicTransparent600,
            color: textBasicColor
        }
    },
    '.ͼb': { color: 'var(--text-primary-focus-color)' },
    '.ͼc': {
        color: textHintColor
    },
    '.ͼd': { color: 'var(--text-warning-focus-color)' },
    '.ͼe': { color: 'var(--text-danger-focus-color)' },
    '.ͼf': { color: 'var(--text-danger-hover-color)' },
    '.ͼg': { color: '#940' },
    '.ͼh': { color: '#708' },
    '.ͼi': { color: 'var(--text-warning-color)' },
    '.ͼj': { color: 'var(--text-success-color)' },
    '.ͼk': { color: 'var(--text-success-focus-color)' },
    '.ͼl': { color: 'var(--text-info-color)' },
    '.ͼm': { color: 'var(--text-primary-color)' },
    '.ͼn': { color: 'var(--text-danger-color)' }
}, { dark: false });
