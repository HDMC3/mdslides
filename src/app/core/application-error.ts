export class ApplicationError extends Error {
    constructor(message: string, public title?: string) {
        super(message);
    }
}
