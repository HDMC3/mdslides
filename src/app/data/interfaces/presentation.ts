import { Slide } from './slide';

export interface Presentation {
    id: string;
    title: string;
    description: string;
    slides: Slide[];
    creation_date: Date;
    modification_date: Date;
}
