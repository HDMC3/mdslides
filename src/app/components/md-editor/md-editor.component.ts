import { Component, ElementRef, OnInit, Output, EventEmitter } from '@angular/core';
import { markdown } from '@codemirror/lang-markdown';
import { basicSetup } from 'codemirror';
import { EditorState } from '@codemirror/state';
import { languages } from '@codemirror/language-data';
import { EditorView } from '@codemirror/view';
import { oneDarkTheme } from './editor-theme';

@Component({
    selector: 'app-md-editor',
    templateUrl: './md-editor.component.html',
    styleUrls: ['./md-editor.component.css']
})
export class MdEditorComponent implements OnInit {
    @Output() valueChange: EventEmitter<string[]>;
    editorInitialized: boolean;

    constructor(private elementRef: ElementRef) {
        this.editorInitialized = false;
        this.valueChange = new EventEmitter();
    }

    ngOnInit(): void {
        const state = EditorState.create({
            extensions: [
                basicSetup,
                markdown({ codeLanguages: languages }),
                EditorView.domEventHandlers({
                    keyup: (e, v) => {
                        const editorValue = v.state.doc.toJSON();
                        this.valueChange.emit(editorValue);
                    }
                }),
                EditorView.lineWrapping,
                oneDarkTheme
            ]
        });

        const view = new EditorView({
            state,
            parent: this.elementRef.nativeElement
        });

        view.dom.style.height = '100%';
    }

    initEditor() {
    }

}
