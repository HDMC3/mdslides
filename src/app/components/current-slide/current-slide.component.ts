import { Component, ElementRef, HostBinding, OnInit, AfterViewInit } from '@angular/core';
import { FormControl, Validators } from '@angular/forms';
import { NbDialogService } from '@nebular/theme';
import { MarpitService } from 'src/app/core/services/marpit.service';
import { MdEditorService } from 'src/app/core/services/md-editor.service';
import { PresentationService } from 'src/app/core/services/presentation.service';
import { MarpThemeData } from 'src/app/core/types/marp-theme-data';
import { Slide } from 'src/app/data/interfaces/slide';
import { FormValidators } from 'src/app/shared/form-validators';
import { MarpitThemesDialogComponent } from '../marpit-themes-dialog/marpit-themes-dialog.component';

@Component({
    selector: 'app-current-slide',
    templateUrl: './current-slide.component.html',
    styleUrls: ['./current-slide.component.css']
})
export class CurrentSlideComponent implements OnInit, AfterViewInit {

    @HostBinding('style.--zoom') componentStyle = window.innerWidth <= 500 ? '0.3' : '0.4';
    editorValue: string[];
    editorValueStr: string;
    editingSlideName: boolean;
    slideNameFormControl: FormControl;
    currentSlide?: Slide;
    currentTheme?: MarpThemeData;

    constructor(
        private mdEditorService: MdEditorService,
        private marpitService: MarpitService,
        private presentationService: PresentationService,
        private componentRef: ElementRef,
        private nbDialogService: NbDialogService
    ) {
        this.editorValue = [];
        this.editorValueStr = '';
        this.editingSlideName = false;
        this.slideNameFormControl = new FormControl('', [Validators.required, FormValidators.noEmpty]);
    }

    ngOnInit(): void {
        this.presentationService.currentSlide$.subscribe(slide => {
            this.currentSlide = slide;
            if (!slide) return;
            this.editingSlideName = false;
            this.slideNameFormControl.setValue(slide.name);
            this.renderSlide(slide.code);
        });

        this.marpitService.currentTheme$.subscribe(theme => {
            this.renderSlide(this.editorValue);
            this.currentTheme = theme;
        });
    }

    ngAfterViewInit() {
        this.mdEditorService.editorValue.subscribe(editorChangeData => {
            const newEditorValueStr = editorChangeData.value.join('');
            if (this.editorValueStr === newEditorValueStr) return;
            this.renderSlide(editorChangeData.value);
            this.editorValueStr = newEditorValueStr;
            this.editorValue = editorChangeData.value;
        });
    }

    zoomSlide(action: 'max' | 'min') {
        const currentZoom = this.componentRef.nativeElement.style.getPropertyValue('--zoom');
        const newZoom = action === 'max' ? Number.parseFloat(currentZoom) + 0.05 : Number.parseFloat(currentZoom) - 0.05;

        if (newZoom > 1 || newZoom <= 0.15) return;

        this.componentRef?.nativeElement.style.setProperty('--zoom', `${newZoom}`);
    }

    renderSlide(editorValue: string[]) {
        const currentSlideContainer = this.componentRef.nativeElement.querySelector('.current-slide-container');
        if (!currentSlideContainer) throw new Error();

        let shadowRoot = currentSlideContainer.shadowRoot;
        if (!shadowRoot) shadowRoot = currentSlideContainer.attachShadow({ mode: 'open' });

        const { html, css } = this.marpitService.render(editorValue);
        shadowRoot.innerHTML = `<style>${css}\n.marpit{position: absolute; top: 50%; left: 50%; transform: translate(-50%, -50%) scale(var(--zoom)); box-shadow: var(--shadow);}</style>\n${html}`;
        shadowRoot.querySelectorAll('section').forEach((el: any) => {
            if (el.id !== '1') el.remove();
        });
    }

    editSlideName() {
        this.editingSlideName = true;
    }

    saveSlideName() {
        if (!this.currentSlide) throw new Error();
        if (this.slideNameFormControl.invalid) return;
        this.currentSlide.name = this.slideNameFormControl.value;
        this.presentationService.updateStorage();
        this.editingSlideName = false;
    }

    openMarpitThemesDialog() {
        this.nbDialogService.open(MarpitThemesDialogComponent, {
            context: {
                currentTheme: this.currentTheme
            }
        });
    }

}
