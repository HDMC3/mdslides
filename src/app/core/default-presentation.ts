import { v4 } from 'uuid';
import { Presentation } from '../data/interfaces/presentation';

export const getDefaultPresentation = (id: string): Presentation => {
    return {
        id,
        title: 'Titulo de presentacion',
        creation_date: new Date(),
        modification_date: new Date(),
        description: '',
        slides: [
            {
                id: v4(),
                name: 'Presentacion inicial',
                code: [
                    '---',
                    'style: |',
                    'section {',
                    '   display: flex;',
                    '   flex-direction: column;',
                    '   align-items: center;',
                    '   justify-content: center;',
                    '}',
                    '---',
                    '',
                    '# Inicio de la presentacion',
                    '# ðŸ˜„âœŒ'
                ]
            }
        ]
    };
};
