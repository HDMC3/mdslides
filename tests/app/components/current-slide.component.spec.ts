import { ComponentFixture, TestBed } from '@angular/core/testing';
import { NbButtonModule, NbDialogModule, NbDialogService, NbIconModule, NbInputModule, NbThemeModule } from '@nebular/theme';


import { MarpitService } from '../../../src/app/core/services/marpit.service';
import { MdEditorService } from '../../../src/app/core/services/md-editor.service';
import { PresentationService } from '../../../src/app/core/services/presentation.service';
import { CurrentSlideComponent } from '../../../src/app/components/current-slide/current-slide.component';
import { MaxLengthPipe } from 'src/app/shared/pipes/max-length.pipe';
import { NbEvaIconsModule } from '@nebular/eva-icons';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';

describe('CurrentSlideComponent', () => {
    let component: CurrentSlideComponent;
    let fixture: ComponentFixture<CurrentSlideComponent>;
    let compiled: HTMLElement;
    let presentationService: PresentationService;

    beforeEach(async() => {
        await TestBed.configureTestingModule({
            declarations: [CurrentSlideComponent, MaxLengthPipe],
            imports: [
                FormsModule,
                ReactiveFormsModule,
                NbThemeModule.forRoot({ name: 'default' }),
                NbDialogModule.forRoot(),
                NbIconModule,
                NbEvaIconsModule,
                NbButtonModule,
                NbInputModule
            ],
            providers: [
                MdEditorService,
                MarpitService,
                PresentationService,
                NbDialogService
            ]
        })
            .compileComponents();
    });

    beforeEach(() => {
        fixture = TestBed.createComponent(CurrentSlideComponent);
        component = fixture.componentInstance;
        presentationService = TestBed.inject(PresentationService);

        fixture.detectChanges();
        compiled = fixture.nativeElement;
    });

    it('should create', () => {
        expect(component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(compiled).toMatchSnapshot();
    });

    it('should load current slide when init component', (done) => {
        presentationService.currentSlide$.subscribe(slide => {
            fixture.detectChanges();
            const slideName = compiled.querySelector('.slide-name-container h6');
            const slideNameContent = slideName?.textContent?.replace(/\.\.\./g, '');
            expect(slide?.name).toMatch(new RegExp(`^${slideNameContent}`));
            done();
        });
    });

    it('should show input for edit current slide name', (done) => {
        presentationService.currentSlide$.subscribe(slide => {
            fixture.detectChanges();
            const editNameButton = compiled.querySelector<HTMLButtonElement>('.slide-name-container button');
            editNameButton?.click();
            fixture.detectChanges();
            const inputNameSlide = compiled.querySelector<HTMLInputElement>('.slide-name-container input');
            expect(inputNameSlide).toBeTruthy();
            done();
        });
    });

    it('should not show slide name either zoom buttons', (done) => {
        presentationService.currentSlide$.subscribe(slide => {
            fixture.detectChanges();
            component.currentSlide = undefined;
            fixture.detectChanges();
            const topOptions = compiled.querySelector('.top-options-container');
            const zoomButtons = compiled.querySelector('.zoom-buttons');
            expect(topOptions).toBeNull();
            expect(zoomButtons).toBeNull();
            done();
        });
    });

    it('should increase "--zoom" style property value', (done) => {
        presentationService.currentSlide$.subscribe(slide => {
            const beforeZoom = compiled.style.getPropertyValue('--zoom');
            const zoomButtons = compiled.querySelectorAll<HTMLButtonElement>('.zoom-buttons button');
            const increaseZoomButton = zoomButtons[1];
            increaseZoomButton.click();
            fixture.detectChanges();
            const afterZoom = compiled.style.getPropertyValue('--zoom');
            expect(Number(beforeZoom)).toBeLessThan(Number(afterZoom));
            done();
        });
    });

    it('should increase "--zoom" style property value', (done) => {
        presentationService.currentSlide$.subscribe(slide => {
            const beforeZoom = compiled.style.getPropertyValue('--zoom');
            const zoomButtons = compiled.querySelectorAll<HTMLButtonElement>('.zoom-buttons button');
            const decreaseZoomButton = zoomButtons[0];
            decreaseZoomButton.click();
            fixture.detectChanges();
            const afterZoom = compiled.style.getPropertyValue('--zoom');
            expect(Number(beforeZoom)).toBeGreaterThan(Number(afterZoom));
            done();
        });
    });

    it('"--zoom" style property value should not be greater than 1', () => {
        fixture.detectChanges();
        for (let i = 0; i < 30; i++) {
            component.zoomSlide('max');
        }
        const afterZoom = compiled.style.getPropertyValue('--zoom');
        expect(Number(afterZoom)).toBeLessThanOrEqual(1);
    });

    it('"--zoom" style property value should not be less than or equal 0.15', () => {
        fixture.detectChanges();
        for (let i = 0; i < 30; i++) {
            component.zoomSlide('min');
        }
        const afterZoom = compiled.style.getPropertyValue('--zoom');
        expect(Number(afterZoom)).toBeGreaterThan(0.15);
    });
});
