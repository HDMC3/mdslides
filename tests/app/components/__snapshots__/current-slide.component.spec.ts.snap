// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentSlideComponent should match snapshot 1`] = `
<div
  id="root1"
  style="--zoom: 0.4;"
>
  <div
    class="current-slide-container"
  />
  <div
    class="top-options-container"
  >
    <div
      class="slide-name-container"
    >
      <button
        aria-disabled="false"
        class="appearance-ghost size-medium shape-round icon-start icon-end status-basic"
        ghost=""
        nbbutton=""
        shape="round"
        tabindex="0"
      >
        <nb-icon
          icon="edit-outline"
        >
          <svg
            class="eva eva-edit-outline"
            fill="currentColor"
            height="100%"
            viewBox="0 0 24 24"
            width="100%"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              data-name="Layer 2"
            >
              <g
                data-name="edit"
              >
                <rect
                  height="24"
                  opacity="0"
                  width="24"
                />
                <path
                  d="M19.4 7.34L16.66 4.6A2 2 0 0 0 14 4.53l-9 9a2 2 0 0 0-.57 1.21L4 18.91a1 1 0 0 0 .29.8A1 1 0 0 0 5 20h.09l4.17-.38a2 2 0 0 0 1.21-.57l9-9a1.92 1.92 0 0 0-.07-2.71zM9.08 17.62l-3 .28.27-3L12 9.32l2.7 2.7zM16 10.68L13.32 8l1.95-2L18 8.73z"
                />
              </g>
            </g>
          </svg>
        </nb-icon>
      </button>
      <h6
        style="margin: 0;"
      >
        Inicio de presentaci√≥n
      </h6>
    </div>
    
    
    <button
      aria-disabled="false"
      class="appearance-ghost size-medium shape-round icon-start icon-end status-basic"
      ghost=""
      nbbutton=""
      shape="round"
      tabindex="0"
    >
      <nb-icon
        icon="color-palette-outline"
      >
        <svg
          class="eva eva-color-palette-outline"
          fill="currentColor"
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="color-palette"
            >
              <rect
                height="24"
                opacity="0"
                width="24"
              />
              <path
                d="M19.54 5.08A10.61 10.61 0 0 0 11.91 2a10 10 0 0 0-.05 20 2.58 2.58 0 0 0 2.53-1.89 2.52 2.52 0 0 0-.57-2.28.5.5 0 0 1 .37-.83h1.65A6.15 6.15 0 0 0 22 11.33a8.48 8.48 0 0 0-2.46-6.25zM15.88 15h-1.65a2.49 2.49 0 0 0-1.87 4.15.49.49 0 0 1 .12.49c-.05.21-.28.34-.59.36a8 8 0 0 1-7.82-9.11A8.1 8.1 0 0 1 11.92 4H12a8.47 8.47 0 0 1 6.1 2.48 6.5 6.5 0 0 1 1.9 4.77A4.17 4.17 0 0 1 15.88 15z"
              />
              <circle
                cx="12"
                cy="6.5"
                r="1.5"
              />
              <path
                d="M15.25 7.2a1.5 1.5 0 1 0 2.05.55 1.5 1.5 0 0 0-2.05-.55z"
              />
              <path
                d="M8.75 7.2a1.5 1.5 0 1 0 .55 2.05 1.5 1.5 0 0 0-.55-2.05z"
              />
              <path
                d="M6.16 11.26a1.5 1.5 0 1 0 2.08.4 1.49 1.49 0 0 0-2.08-.4z"
              />
            </g>
          </g>
        </svg>
      </nb-icon>
    </button>
  </div>
  
  <div
    class="zoom-buttons"
  >
    <button
      aria-disabled="false"
      class="appearance-outline size-medium shape-rectangle icon-start icon-end status-basic"
      nbbutton=""
      outline=""
      size="medium"
      tabindex="0"
    >
      <nb-icon
        icon="minimize-outline"
      >
        <svg
          class="eva eva-minimize-outline"
          fill="currentColor"
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="minimize"
            >
              <rect
                height="24"
                opacity="0"
                width="24"
              />
              <path
                d="M20.71 19.29l-3.4-3.39A7.92 7.92 0 0 0 19 11a8 8 0 1 0-8 8 7.92 7.92 0 0 0 4.9-1.69l3.39 3.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 11a6 6 0 1 1 6 6 6 6 0 0 1-6-6z"
              />
              <path
                d="M13 10H9a1 1 0 0 0 0 2h4a1 1 0 0 0 0-2z"
              />
            </g>
          </g>
        </svg>
      </nb-icon>
    </button>
    <button
      aria-disabled="false"
      class="appearance-outline size-medium shape-rectangle icon-start icon-end status-basic"
      nbbutton=""
      outline=""
      size="medium"
      tabindex="0"
    >
      <nb-icon
        icon="maximize-outline"
      >
        <svg
          class="eva eva-maximize-outline"
          fill="currentColor"
          height="100%"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            data-name="Layer 2"
          >
            <g
              data-name="maximize"
            >
              <rect
                height="24"
                opacity="0"
                width="24"
              />
              <path
                d="M20.71 19.29l-3.4-3.39A7.92 7.92 0 0 0 19 11a8 8 0 1 0-8 8 7.92 7.92 0 0 0 4.9-1.69l3.39 3.4a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42zM5 11a6 6 0 1 1 6 6 6 6 0 0 1-6-6z"
              />
              <path
                d="M13 10h-1V9a1 1 0 0 0-2 0v1H9a1 1 0 0 0 0 2h1v1a1 1 0 0 0 2 0v-1h1a1 1 0 0 0 0-2z"
              />
            </g>
          </g>
        </svg>
      </nb-icon>
    </button>
  </div>
  
</div>
`;
